### ISSUE TITLE
Collection filtering

### ISSUE THREAD

**Issue by aazuspan** [darrenwiens/stac-nb#6]
Hey @darrenwiens, what do you think about having a text box in the Common tab that filters collections by ID, to help navigate big STACs? Whenever the filter text is changed, the collection list would update to only include matching collections. 

I threw together a quick working demo below, mostly to make sure it was possible with `ipywidgets` before I suggested it.

If you like the idea I'm happy to make a PR. If not, no worries!

![stacnb_filter](https://user-images.githubusercontent.com/50475791/164958881-a6a02642-d2c3-4d54-bec1-e93e66385e5f.gif)

**Comment by darrenwiens** [comment]
Sorry for the delay, I like the idea for this added functionality!

The other inputs (`Collection(s)`, `Limit`, and `ID(s)`) all act on the `query`, but the new filter acts on `Collection(s)`. Is there a way to group `Collection(s)` together with `Filter` to make it more clear from a UX perspective?

**Comment by aazuspan** [comment]
Good point, I played around with some changes to try and make it clearer that the filter doesn't affect the query.

1. Changed `Filter` to 🔍 so it feels less like a query input.
2. Added `# selected` so you can tell the filter input doesn't select anything.
3. Divided into `Collections` and `Items` sections.

![demo with headers](https://user-images.githubusercontent.com/50475791/198851802-6d6b8270-d565-4000-82c3-0a6b1d63aad4.gif)

\# 3 might be overkill, but I figured I'd throw some options your way.

**Comment by darrenwiens** [comment]
That looks awesome! Happy to review a PR.
